File: lc_py/codeforces/greedy/discount.py
import sys

def solve(prices, voucher):
    prices.sort()
    voucher.sort()
    s = sum(prices)
    nn = len(prices)
    for vi in voucher:
        if nn < vi: break
        s -= prices[nn - vi]
        nn -= vi
    return s

if __name__ == "__main__":
    input = sys.stdin.read
    data = input().split()
    t = int(data[0])
    idx = 1
    for _ in range(t):
        n = int(data[idx])
        k = int(data[idx + 1])
        idx += 2
        prices = [int(x) for x in data[idx:idx + n]]
        idx += n
        vouchers = [int(x) for x in data[idx:idx + k]]
        idx += k
        print(solve(prices, vouchers))

File: lc_go/codeforces/ds/greedy/discount.go
package main

import (
	"fmt"
	"sort"
)

func solve(prices []int64, vouchers []int) int64 {
	sort.Slice(prices, func(i, j int) bool { return prices[i] < prices[j] })
	sort.Ints(vouchers)
	s := int64(0)
	for _, x := range prices {
		s += x
	}
	nn := len(prices)
	for _, vi := range vouchers {
		if nn < vi {
			break
		}
		s -= prices[nn-vi]
		nn -= vi
	}
	return s
}

func main() {
	var t, n, k int
	fmt.Scan(&t)
	for ; t > 0; t-- {
		fmt.Scan(&n, &k)
		prices := make([]int64, n)
		for i := 0; i < n; i++ {
			fmt.Scan(&prices[i])
		}
		vouchers := make([]int, k)
		for i := 0; i < k; i++ {
			fmt.Scan(&vouchers[i])
		}
		fmt.Println(solve(prices, vouchers))
	}
}
